src/main/java/appeng/util/inv/ItemHandlerIterator.java

```diff
@@ -0,0 +1,39 @@
+package appeng.util.inv;
+
+
+import java.util.Iterator;
+
+import net.minecraftforge.items.IItemHandler;
+
+
+class ItemHandlerIterator implements Iterator<ItemSlot>
+{
+
+	private final IItemHandler itemHandler;
+
+	private final ItemSlot itemSlot = new ItemSlot();
+
+	private int slot = 0;
+
+	public ItemHandlerIterator( IItemHandler itemHandler )
+	{
+		this.itemHandler = itemHandler;
+	}
+
+	@Override
+	public boolean hasNext()
+	{
+		return slot < itemHandler.getSlots();
+	}
+
+	@Override
+	public ItemSlot next()
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Add a "NoSuchElementException" for iteration beyond the end of the collection. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2272)